targets/XPlatCppSdk/source/code/source/playfab/PlayFabIOSHttpPlugin.mm

```diff
@@ -0,0 +1,346 @@
+#include <stdafx.h>
+#import <UIKit/UIKit.h>
+
+#include <playfab/PlayFabIOSHttpPlugin.h>
+#include <playfab/PlayFabSettings.h>
+#include <stdexcept>
+#include <vector>
+
+namespace PlayFab
+{
+    struct PlayFabIOSHttpPlugin::RequestImpl
+    {
+        RequestImpl() :
+            sessionTask(nil)
+        {
+
+        }
+
+        virtual ~RequestImpl()
+        {
+            sessionTask = nil;
+        }
+
+        void Cancel()
+        {
+            if(sessionTask)
+            {
+                [sessionTask cancel];
+            }
+        }
+
+        __strong NSURLSessionUploadTask* sessionTask;
+    };
+
+}
+
+namespace PlayFab
+{
+    PlayFabIOSHttpPlugin::RequestTask::RequestTask() :
+        state(RequestTask::None),
+        impl(nullptr),
+        requestContainer(nullptr)
+    {
+
+    }
+
+    PlayFabIOSHttpPlugin::RequestTask::~RequestTask()
+    {
+        if(impl)
+        {
+            delete impl;
+            impl = nullptr;
+        }
+    }
+
+    bool PlayFabIOSHttpPlugin::RequestTask::Initialize(std::unique_ptr<CallRequestContainerBase>& requestContainer)
+    {
+        this->requestContainer = std::move(requestContainer);
+        this->impl = new PlayFabIOSHttpPlugin::RequestImpl();
+        return true;
+    }
+    void PlayFabIOSHttpPlugin::RequestTask::Cancel()
+    {
+        if(this->impl)
+        {
+            this->impl->Cancel();
+        }
+    }
+
+    
+}
+
+namespace PlayFab
+{
+    PlayFabIOSHttpPlugin::PlayFabIOSHttpPlugin() :
+        workerThread(nullptr),
+        threadRunning(false),
+        requestingTask(nullptr)
+    {
+
+    }
+
+    PlayFabIOSHttpPlugin::~PlayFabIOSHttpPlugin()
+    {
+        threadRunning = false;
+
+        httpRequestMutex.lock();
+        if(this->requestingTask)
+        {
+            this->requestingTask->Cancel();
+        }
+        if(workerThread)
+        {
+            httpRequestMutex.unlock();
+            workerThread->join();
+            httpRequestMutex.lock();
+        }
+        httpRequestMutex.unlock();
+    }
+
+    void PlayFabIOSHttpPlugin::MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer)
+    {
+        std::shared_ptr<RequestTask> requestTask = nullptr;
+        try
+        {
+            requestTask = std::make_shared<RequestTask>();
+            requestTask->Initialize(requestContainer);
+        }
+        catch (const std::exception& ex)
+        {
+            PlayFabPluginManager::GetInstance().HandleException(ex);
+        }
+        catch (...)
+        {
+
+        }
+        if(requestTask != nullptr)
+        { // LOCK httpRequestMutex
+            std::unique_lock<std::mutex> lock(httpRequestMutex);
+            requestTask->state = RequestTask::Pending;
+            pendingRequests.push_back(std::move(requestTask));
```

Comment:
>pendingRequests.push_back(std::move(requestTask)); [](start = 12, length = 50)

isn't it better to make post request and then add to the queue only to process result? if current SDK design is same then its ok, otherwise this is something to keep in mind for future.